metaflow/plugins/aip/argo_utils.py

```diff
@@ -14,16 +14,38 @@
 
 def run_argo_workflow(
     kubernetes_namespace: str,
-    template_name: str,
+    template_name: Optional[str] = None,
+    project_name: Optional[str] = None,
+    branch_name: Optional[str] = None,
+    template_name_prefix: Optional[str] = None,
     parameters: Optional[dict] = None,
     wait_timeout: Union[int, float, datetime.timedelta] = 0,
     **kwarg,  # Other parameters for wait function
 ) -> Tuple[str, str]:
+    """
+    Using template_name to trigger a workflow template name with exact match.
+
+    If no template_name is provided, the latest workflow template satisfying
+    the project_name, branch_name and template_name_prefix will be used.
+    All of these filters are optional. If not provided, the latest workflow
+    from the namespace will be used.
+    """
+    client = ArgoClient(namespace=kubernetes_namespace)
+
+    if not template_name:
```

Comment:
should we raise an error if `template_name_prefix` and `template_name` is empty?

Reply:
Debatable. We can raise a warning that some filter should be used. So the users may

- Exact match using `template_name`
- Prefix match with `template_name_prefix` (and optionally filter using labels)
- Only filter using the labels without `template_name_prefix`

For the 3rd case here, both `template_name_prefix` and `template_name` can be empty. I'll update to include a warning.

Reply:
Added the error.